### ISSUE TITLE
a few issues when dealing with ULT data

### ISSUE THREAD

**Issue by jonorthwash** [SwatPhonLab/UltraTrace#128]
Apparent regressions:
- [x] Window resize doesn't resize loaded US frame, spectrogram, or tiers.
- [ ] Click-and-drag on spectrogram to select span of audio appears to be broken.
- [x] Playback doesn't seem to work.

Other stuff:
- [x] Trace set not selected by default
- [x] "Read scan line data" should be selected (and data loaded) by default if it's the only option.
- [ ] It's possible to go out of bounds with the TextGrid.  E.g., advancing frames with the arrow key past the last frame results in `IndexError: tuple index out of range`

These can be made separate issues if desired.

**Comment by jonorthwash** [comment]
- [ ] Sometimes the first frame is shown as being in the selected interval when the last interval is selected.  It also can't be clicked on.
![Screenshot from 2020-01-02 15-29-22](https://user-images.githubusercontent.com/963849/71691181-a96a6580-2d74-11ea-8a7d-3be61ddcb0f5.png)

**Comment by jonorthwash** [comment]
- [x] Figure out what's up with the extra set of traces called "null" being added.

**Comment by jonorthwash** [comment]
>  * [x]  Figure out what's up with the extra set of traces called "null" being added.

This apparently was left over from the alignment issue.  Deleting `metadata.json` and running again seems to've fixed it.

**Comment by jonorthwash** [comment]
- [ ] Are frames being overwritten?
```
INFO:   Line 21 of SPLINES.dat imported as File010 frame 99
INFO:   Line 22 of SPLINES.dat imported as File010 frame 100
INFO:   Line 23 of SPLINES.dat imported as File010 frame 99
INFO:   Line 24 of SPLINES.dat imported as File010 frame 100
INFO:   Line 25 of SPLINES.dat imported as File010 frame 99
INFO:   Line 26 of SPLINES.dat imported as File010 frame 100
```
- [ ] Why are some lines unable to be imported?
```
WARN:   Unable to import line 71 of SPLINES.dat
WARN:   Unable to import line 72 of SPLINES.dat
WARN:   Unable to import line 73 of SPLINES.dat
```

**Comment by mr-martian** [comment]
> * [ ]  Are frames being overwritten?
> 
> ```
> INFO:   Line 21 of SPLINES.dat imported as File010 frame 99
> INFO:   Line 22 of SPLINES.dat imported as File010 frame 100
> INFO:   Line 23 of SPLINES.dat imported as File010 frame 99
> INFO:   Line 24 of SPLINES.dat imported as File010 frame 100
> INFO:   Line 25 of SPLINES.dat imported as File010 frame 99
> INFO:   Line 26 of SPLINES.dat imported as File010 frame 100
> ```
> 

Yes, because that `SPLINES.dat` has 3 consecutive pairs of lines with the same date. They have different annotations though, perhaps it should try to load the textgrid and check whether the label matches?

> * [ ]  Why are some lines unable to be imported?
> 
> ```
> WARN:   Unable to import line 71 of SPLINES.dat
> WARN:   Unable to import line 72 of SPLINES.dat
> WARN:   Unable to import line 73 of SPLINES.dat
> ```

Because the `File030.ult`'s mime type comes up as `application/x-dosexec`, so it's ignored and then the importer won't be able to compute frame times so it skips File030. Fixed in 7c6fec5.

**Comment by jonorthwash** [comment]
> Yes, because that `SPLINES.dat` has 3 consecutive pairs of lines with the same date.

That in and of itself isn't a problem.  I guess the issue here is the second field, "time of sample in recording".  These lines, besides their annotation label, appear to be legit repeats.  I wonder why the AAA export did that....

> They have different annotations though, perhaps it should try to load the textgrid and check whether the label matches?

Yeah, it would be good to try to match the label to whatever's in the TextGrid at that time index.  Normally throw a warning for non-matches, and when there are conflicts, go with either the correct one or the last(?), or both if there are multiple repeats (like their are here).

As long as we load the right data for this file, it's not very important.
